kolibri_explore_plugin/assets/src/searchApi.js

```diff
@@ -0,0 +1,56 @@
+import { KolibriApi } from './kolibriApi';
+
+function zimSearch(zimfile, query, node, maxResults = 10, start) {
+  const data = {
+    query,
+    max_results: maxResults,
+  };
+  if (start) {
+    data.start = start;
+  }
+  const { origin } = window.location;
+  const url = new URL(`${origin}/zim/search/${zimfile.checksum}.zim`);
+  for (const k in data) {
+    url.searchParams.append(k, data[k]);
+  }
+  return fetch(url)
+    .then(r => r.json())
+    .then(r => ({ ...r, node }));
+}
+
+function zimToNode(zimArticle, node) {
+  return {
+    ...zimArticle,
+    node: node,
+    id: `${node.id}${zimArticle.path}`,
+    nodeId: node.id,
+    channelId: node.channel_id,
+    description: zimArticle.snippet,
+    kind: 'zim',
+    directContent: true,
+    nodeUrl: `/topics/${node.channel_id}/c/${node.id}?zimPath=${zimArticle.path}`,
+  };
+}
+
+export function wikiChannelSearch(channelId, query) {
+  const kolibri = new KolibriApi(channelId);
+  // Get all the zim nodes in the channel
+  return kolibri
+    .getContentByFilter({
+      kinds: ['zim'],
+    })
+    .then(({ results }) => {
+      // Do the search for each zim node
+      const promises = results.map(node => {
+        const [zimfile] = node.files.filter(f => f.extension === 'zim');
+        return zimSearch(zimfile, query, node);
+      });
+      return Promise.all(promises);
```

Comment:
You can return `node` here along with the promise returned by `Promise.all`. And avoid passing it to the `zimSearch` function.

Reply:
every call to `zimSearch` has a different `node`. In any case I'll try to avoid to pass the node to the `zimSearch` function.

Reply:
I think you don't understand me but no problem, this whole search stuff needs a refactor anyways.